lib/sles4sap/ipaddr2.pm

```diff
@@ -236,6 +242,19 @@ sub ipaddr2_azure_deployment {
         frontend_ip => $lb_fe,
         name => $lb . "_rule",
         port => '80');
+
+    if ($args{diagnostic} eq 1) {
```

Comment:
Can't it also be 
`if ($args{diagnostic})`
?

Reply:
Agree with Bill. `eq 1` will:

- Cast `$args{diagnostic}` into a string.
- Cast `1` into a string.
- Do a string equal check.

So, `if ($args{diagnostic}` will already do a conversion to boolean and evaluate it which is a bit better.